dataregistry-extension/src/csvviewer/data_grid.ts

```diff
@@ -0,0 +1,76 @@
+/**
+ * @license BSD-3-Clause
+ *
+ * Copyright (c) 2019 Project Jupyter Contributors.
+ * Distributed under the terms of the 3-Clause BSD License.
+ */
+
+import { Observable, Subscription } from "rxjs";
+import { DataGrid } from "@phosphor/datagrid";
+import { Message } from "@phosphor/messaging";
+import { DSVModel } from "@jupyterlab/csvviewer";
+
+/**
+ * Class to support the conversion from CSV data to `DataGrid` data.
+ *
+ * @private
+ */
+class CSVDataGrid extends DataGrid {
+  private _data: Observable<string>;
+  private _subscription?: Subscription;
+
+  /**
+   * Constructor.
+   *
+   * @param data - CSV data as an observable
+   * @returns class instance
+   */
+  constructor(data: Observable<string>) {
+    super();
+    this._data = data;
+    this.headerVisibility = "all";
+  }
+
+  /**
+   * Callback invoked upon receiving a `'before-attach'` message.
+   *
+   * @param msg - message
+   */
+  protected onBeforeAttach(msg: Message) {
+    const self = this;
+    this._subscription = this._data.subscribe(onData);
+    super.onBeforeAttach(msg);
+
+    /**
+     * Callback invoked upon changes to the CSV data.
+     *
+     * @private
+     * @param data - CSV data
+     */
+    function onData(data: string) {
```

Comment:
Maybe make this a method?

Reply:
Yes, this could make sense. Generally, I avoid hanging private (implementation) methods off prototypes to ensure encapsulation, unless their are clear performance advantages to do otherwise.
​
Which raises the question: how frequently is the `before-attach` message sent? If often, then I should definitely create an private internal method; but, if not, then probably fine to leave as is.